src/write_stage.rs

```diff
@@ -38,32 +44,97 @@ impl WriteStage {
         let entries = entry_receiver.recv_timeout(Duration::new(1, 0))?;
         let votes = entries_to_votes(&entries);
         crdt.write().unwrap().insert_votes(&votes);
+
+        //TODO(anatoly): real stake based voting needs to change this
+        //leader simply votes if the current set of validators have voted
+        //on a valid last id
         entry_writer.write_and_register_entries(&entries)?;
         trace!("New blobs? {}", entries.len());
         let mut blobs = VecDeque::new();
         entries.to_blobs(blob_recycler, &mut blobs);
+
         if !blobs.is_empty() {
-            inc_new_counter!("write_stage-broadcast_vote-count", votes.len());
-            inc_new_counter!("write_stage-broadcast_blobs-count", blobs.len());
+            inc_new_counter!("write_stage-recv_vote", votes.len());
+            inc_new_counter!("write_stage-broadcast_blobs", blobs.len());
             trace!("broadcasting {}", blobs.len());
             blob_sender.send(blobs)?;
         }
         Ok(())
     }
-
+    pub fn leader_vote(
```

Comment:
Can you explain a bit what this function is and why it's in WriteStage? I'm behind on the latest codebase changes, but see that there's a VoteStage now. What's the relationship?

And can you rename the function to something in the form `<verb>_<subject>`?

Reply:
VoteStage is for validators, this is logic for leaders. I assume it's here because we wanted to do it after the leader had finalized on an id in the ledger. I'll rename it.